pkg/KubeArmorController/main.go

```diff
@@ -139,29 +143,29 @@ func main() {
 }
 
 // detect the enforcer on the node
-func detectEnforcer(logger logr.Logger) string {
+func detectEnforcer(logger logr.Logger, client client.Client, nodeName string) string {
```

Comment:
We can use `context.Context` here instead of using `context.TODO()` later, but I think it's okay given we use `context.TODO()` at other places in the code as well. 
// cc @daemon1024 

w.r.t this change, please update the call where we are registering webhook with updated arguments.
We should update this `Enforcer: detectEnforcer(setupLog),` as well.

Reply:
@kranurag7 should I make changes in podannotator struct itself or just add client anf node name in the same file's  main function